apps/web/src/app/monitor/[id]/page.tsx

```diff
@@ -1,8 +1,33 @@
+import * as z from "zod";
+
+import { availableRegions } from "@openstatus/tinybird";
+
 import { columns } from "@/components/monitor/columns";
 import { DataTable } from "@/components/monitor/data-table";
 import { getResponseListData } from "@/lib/tb";
 
-export default async function Monitor({ params }: { params: { id: string } }) {
-  const data = await getResponseListData({ siteId: params.id });
+//
+
+/**
+ * allowed URL search params
+ */
+const searchParamsSchema = z.object({
+  statusCode: z.coerce.number().optional(),
+  region: z.enum(availableRegions).optional(),
+  cronTimestamp: z.coerce.number().optional(),
+});
+
+export default async function Monitor({
+  params,
+  searchParams,
+}: {
+  params: { id: string };
+  searchParams: { [key: string]: string | string[] | undefined };
```

Comment:
We can use z.infer<typeof paramsschema> ?

Reply:
We shouldn't trust users input. Anywho can modify them. `?hello=world` would bypass it. That's why we validate schema on SSR.